attestation/context.go

```diff
@@ -205,48 +165,34 @@ func (ctx *AttestationContext) runAttestor(attestor Attestor) error {
 		ctx.addMaterials(materialer)
 	}
 
-	if producter, ok := attestor.(Producer); ok {
-		ctx.addProducts(producter)
+	if producer, ok := attestor.(Producer); ok {
+		ctx.addProducts(producer)
 	}
 
 	return nil
 }
 
 func (ctx *AttestationContext) CompletedAttestors() []CompletedAttestor {
-	attestors := make([]CompletedAttestor, len(ctx.completedAttestors))
-	copy(attestors, ctx.completedAttestors)
-	return attestors
+	return ctx.completedAttestors
```

Comment:
The thought process behind the `copy`s of this slice was to prevent someone from mucking about with the AttestationContext's internal state. For instance, by returning this my question is if we can now inadvertently add another Attestor to the backing array of `ctx.completedAttestors`

However, upon playing with this in the playground, it looks like these copys are entirely unnecessary and I was overthinking it  :) https://go.dev/play/p/iP1efJFDgzG

Reply:
Ope, actually we should still copy these before returning. While appending doesn't affect the array in the struct, you can directly replace elements in the slice:

https://go.dev/play/p/ioXYWyituKW